app/routes/_auth+/reset-password.tsx

```diff
@@ -70,10 +71,16 @@ export async function action({ context, request }: ActionFunctionArgs) {
 
         const authSession = await refreshAccessToken(refreshToken);
 
-        await updateAccountPassword(authSession.userId, password);
+        await updateAccountPassword(
+          authSession.userId,
+          password,
+          authSession.accessToken
+        );
+        //on updating the password. it is removing the session_id from the collection. so we need to create a new session
```

Comment:
@rajeshj11 I don't see anything in supabase docs about this. Where does this information come from? Based on how `signOut` works with others, the user current session will not be destroyed.

Reply:
@rajeshj11 I am not sure what happened here.
Seems like you removed the code that destroys the session.
Can you please address the question that both me and Rapha asked?

Reply:
> Are we sure about that? Even with using `others`?

With signout(token, 'others') it will sign out other sessions belonging to that particular user except the current session.
post updating the password with updateAccountPassword, it will clear the current session due to security reasons(which is a standard practice followed by many companies.)

Reply:
I understand what you mean, and I agree it is best practice to invalidate all user sessions and recreate a new one in the current session that updates the password.
This behavior could prevent a “session copy attack” (so, changing the password without invalidating the whole session would be useless since the stolen token could be used with the duplicated refresh token to obtain a valid access token).

But, if we really want this behavior, we should sign out all the sessions (so, something other than `others`), including the current one, and then the `newSession` makes sense.

So, it is up to you, @DonKoko: Revoke all the sessions or remove the following code.

I think this is why we didn't understand it at first. Now I see what you are trying to achieve.